src/Umbraco.Tests.Common/TestHelperBase.cs

```diff
@@ -86,7 +87,11 @@ public IIOHelper IOHelper
             get
             {
                 if (_ioHelper == null)
-                    _ioHelper = new IOHelper(GetHostingEnvironment());
+                {
+                    var hostingEnvironment = GetHostingEnvironment();
+                    _ioHelper = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? new IOHelperLinux(hostingEnvironment)
+                        : (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? (IIOHelper)new IOHelperOSX(hostingEnvironment) : new IOHelperWindows(hostingEnvironment));
```

Comment:
That's a lot of ternaries, would read a lot nicer if it wern't.

Reply:
I would personally create a dedicated class for environmental properties. `TestEnvironment`

``` c#
public static bool IsLinux => RuntimeInformation.IsOSPlatform(OSPlatform.Linux);

public  static bool IsOSX => RuntimeInformation.IsOSPlatform(OSPlatform.OSX);

public  static bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```